packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -319,8 +322,21 @@ export const getCartDetails = payload => {
 
             const shouldResetCart = !error.networkError && isInvalidCart(error);
             if (shouldResetCart) {
-                // Delete the cached ID from local storage.
-                await dispatch(removeCart());
+                if (isSignedIn) {
+                    // Since simple persistence just deletes auth token without
+                    // informing Redux, we need to perform the sign out action
+                    // to reset the store back to initial state here.
+                    await dispatch(signOut());
+                } else {
+                    // Delete the cached ID from local storage.
+                    await dispatch(removeCart());
```

Comment:
I was a little surprised to see this as an `else`. I'm assuming both `removeCart` and `signOut` delete the auth token from local storage?

Reply:
Ok nevermind, just read `user/asyncActions`. Might be useful to add that `removeCart` is called as part of `signOut` to the comment here.

Reply:
I'm going to investigate this more; this change should be resetting customer global state back to initial guest values, so if there are still components rendering like you're logged in, there's a bug somewhere.

Will add that comment and resolve the one area I've observed stale data (left drawer is still pinned like you're logged in). Did you see any other components still rendering with customer data?

Reply:
That was the one I noticed too. No others I think.

Reply:
Fixed in ae8fe42